### PR TITLE
feat: add a modal provider

### PR DESCRIPTION
**Pull request by nicosantangelo**
TODO:
- [x] Fix dependency error with semantics Modal
- [x] Update README

Related: https://github.com/decentraland/ui/pull/26

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1001,6 +1002,95 @@ This way you can change the default locations to use different ones. This will b
...
 </details>
 
+## Modal
+
+Leverages redux state and provides actions to open and close each modal by name. It provides two simple actions:
+
+```ts
+openModal(name: string, metadata: any = null)
+closeModal(name: string)
+closeAllModal()
```
Comment (cazala) [inline README.md]: closeAllModal`s`

File: src/containers/App/App.tsx [hunk 2]
```diff
@@ -13,13 +13,13 @@ export default class App extends React.PureComponent<
     const { hero, isHomePage, children, ...rest } = this.props
     const hasHero = hero != null && isHomePage
     return (
-      <>
+      <React.Fragment>
```
Comment (cazala) [inline src/containers/App/App.tsx]: Why this change?

File: README.md (hunk 1 above)
Comment (nicosantangelo) [reply README.md]: Darn it!

File: src/containers/App/App.tsx (hunk 2 above)
Comment (nicosantangelo) [reply src/containers/App/App.tsx]: The only other piece of code I found using a Fragment used `React.Fragment` (TranslationProvider.tsx) so I went with that on ModalProvider and wanted to use the same thing everywhere

BUT

looking closer the README uses `<>` 😅 soo...which one do you like best? I'll update the PR only using one of the two. I have no preference

Comment (cazala) [reply src/containers/App/App.tsx]: I would rather use `<>`, i think the marketplace is the only one that still uses `React.Fragment` but everywhere else i think we are using the short version (i like it better 😇 ).

Comment (nicosantangelo) [reply src/containers/App/App.tsx]: <> it is 🚂 !
